clippy_lints/src/copies.rs

```diff
@@ -290,6 +290,14 @@ fn lint_same_then_else<'tcx>(
     }
 }
 
+/// The return tuple is structured as follows:
+/// 1. The amount of equal statements from the start
+/// 2. The amount of equal statements from the end
+/// 3. An indication if the block expressions are the same. This will also be true if both are
+/// `None`
```

Comment:
Why not creating a struct for this? If you have to explain the return type it should probably be a struct.

Reply:
Good point, I'm sometimes not used to the Rust way when it comes to this ^^. The function now returns a struct which seems way cleaner :)